### PR TITLE
Add support to dismiss notifications

### PR DESCRIPTION
**Pull request by humrochagf**

### CODE REVIEW

File: activities/models/timeline_event.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ class Types(models.TextChoices):
 
     published = models.DateTimeField(default=timezone.now)
     seen = models.BooleanField(default=False)
+    dismissed = models.BooleanField(default=False)
```
Comment (andrewgodwin) [inline activities/models/timeline_event.py:58]: We might need to consider an index on this in future, but for now I think it's selective enough it should be fine.

File: users/migrations/0021_remove_domain_ix_domain_state_next.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("users", "0020_alter_identity_local"),
+    ]
+
+    operations = [
+        migrations.RemoveIndex(
+            model_name="domain",
+            name="ix_domain_state_next",
```
Comment (andrewgodwin) [inline users/migrations/0021_remove_domain_ix_domain_state_next.py:15]: This migration should not be here - please delete the migration file (the most recent `main` commit fixes the bug that makes this appear)

Comment (humrochagf) [reply users/migrations/0021_remove_domain_ix_domain_state_next.py]: Oups, forgot to leave it out the commit 😜 
Done

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
Thanks!

**Pull request merged**